lib/engine/game/g_18_esp/step/lay_tile_check.rb

```diff
@@ -12,7 +12,17 @@ def potential_tiles(entity, hex)
   end
 
   def legal_tile_rotation?(entity, hex, tile)
-    hex.tile.towns.none?(&:halt?) ? super : halt_upgrade_legal_rotation?(entity, hex, tile)
+    if hex.tile.towns.none?(&:halt?)
+      return super unless hex.id == @game.class::ARANJUEZ_HEX
+
+      # handle aranjues hex situation. Tile must be connected to madrid.
+      super && tile.exits.any? do |exit|
+        neighbor = hex.neighbors[exit]
+        neighbor&.id == 'F24'
```

Comment:
Do you want a constant here? A pattern I like is:

def aranjues_hex()
  @aranjues_hex ||= @game.hex_by_id('F24')
end

....
super && tile.exits.any? { |exit| hex.neighbors[exit] == aranjues_hex }

Reply:
there are other places in the code where i check if hex.id == "F24",
I created a constant in game.rb, and updated all the places where this is used. Thanks